addons/binding/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/internal/Mcp23017HandlerFactory.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mcp23017.internal;
+
+import static org.openhab.binding.mcp23017.Mcp23017BindingConstants.*;
+
+import java.util.List;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.mcp23017.handler.Mcp23017ContactHandler;
+import org.openhab.binding.mcp23017.handler.Mcp23017SwitchHandler;
+
+import com.google.common.collect.Lists;
+
+/**
+ * The {@link Mcp23017HandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class Mcp23017HandlerFactory extends BaseThingHandlerFactory {
+
+    private final static List<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Lists.newArrayList(THING_TYPE_MCP23017_CONTACT,
```

Comment:
Non-preferred order of modifiers (It was like this in a template, which has been changed recently) correct order is `public static final` See also: http://stackoverflow.com/questions/16731240/what-is-a-reasonable-order-of-java-modifiers-abstract-final-public-static-e#16732059